pylint/checkers/refactoring/recommendation_checker.py

```diff
@@ -210,3 +219,85 @@ def visit_comprehension(self, node: astroid.Comprehension) -> None:
 
                 self.add_message("consider-using-dict-items", node=node)
                 return
+
+    @utils.check_messages("consider-using-str-partition")
+    def visit_subscript(self, node: astroid.Subscript) -> None:
+        """Add message when accessing first or last elements of a str.split()."""
+        assignment = None
+        subscripted_object = node.value
+        if isinstance(subscripted_object, astroid.Name):
+            # Check assignment of this subscripted object
+            assignment_lookup_results = node.value.lookup(node.value.name)
+            if (
+                len(assignment_lookup_results) == 0
+                or len(assignment_lookup_results[-1]) == 0
+            ):
+                return
+            assign_name = assignment_lookup_results[-1][-1]
+            if not isinstance(assign_name, astroid.AssignName):
+                return
+            assignment = assign_name.parent
+            if not isinstance(assignment, astroid.Assign):
+                return
+            # Set subscripted_object to the assignment RHS
+            subscripted_object = assignment.value
+
+        if isinstance(subscripted_object, astroid.Call):
+            # Check if call is .split() or .rsplit()
+            if isinstance(
+                subscripted_object.func, astroid.Attribute
+            ) and subscripted_object.func.attrname in ["split", "rsplit"]:
```

Comment:
```suggestion
            ) and subscripted_object.func.attrname in ("split", "rsplit"):
```
Tuples are more efficient if you know that you don't need to add/remove any elements.

Reply:
Cool! didn't think of that, thanks for pointing that out